src/js/actions/layers.js

```diff
@@ -740,13 +740,19 @@ define(function (require, exports) {
                 if (firstUninitializedLayers.size === 50) {
                     this.whenIdle(initializeLayersBackground, document);
                 }
+            })
+            .error(function (err) {
+                /* jshint multistr: true */
+                log.error("InitializeLayersBackground action failed for document %s, \
+                    it was likely closed by Photoshop - %s", document.name, err.message);
             });
     };
     initializeLayersBackground.action = {
         reads: [locks.JS_DOC],
         writes: [],
         transfers: [initializeLayers],
-        modal: true
+        modal: true,
+        noResetOnFailure: true
```

Comment:
How about calling this just `allowFailure`? I think it's better not to try to describe exactly what will happen in this case in case we want to adjust it somewhat later on.